### ISSUE TITLE
warning C4201: nonstandard extension used: nameless struct/union

### ISSUE THREAD

**Issue by MathiasMagnus** [KhronosGroup/OpenCL-Headers#34]
I guess this is the same issue as was the case with GCC inside `typedef struct _cl_image_desc` for eg. which was circumvented with the code:
```c++
#ifdef __GNUC__
    __extension__   /* Prevents warnings about anonymous union in -pedantic builds */
#endif
    union {
      cl_mem                  buffer;
      cl_mem                  mem_object;
    };
```
MSVC has exactly the same problem when compiled with [/Za, strict ANSI conformance](https://docs.microsoft.com/hu-hu/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4201?view=vs-2017).

**Comment by MathiasMagnus** [comment]
By the way, would giving this enum a name break the ABI? Or are there other implications? Isn't it more elegant to actually conform to ANSI C than to disable extensions?

**Comment by jrprice** [comment]
It wouldn't break the ABI, but it would potentially break existing code, which is undesirable. We'll certainly consider avoiding anonymous unions like this in the future though.

I've merged PR #35, so closing this issue now.
